app/src/main/java/com/example/cm/ui/meetup/MeetupListFragment.java

```diff
@@ -1,80 +1,53 @@
 package com.example.cm.ui.meetup;
 
+import android.annotation.SuppressLint;
 import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
 
 import androidx.annotation.NonNull;
 import androidx.fragment.app.Fragment;
 import androidx.lifecycle.ViewModelProvider;
-import androidx.navigation.Navigation;
-import androidx.recyclerview.widget.LinearLayoutManager;
-
-import android.view.LayoutInflater;
-import android.view.Menu;
-import android.view.MenuInflater;
-import android.view.MenuItem;
-import android.view.View;
-import android.view.ViewGroup;
-import android.widget.ArrayAdapter;
+import androidx.recyclerview.widget.GridLayoutManager;
 
-import com.example.cm.R;
-import com.example.cm.data.repositories.MeetupRepository;
-import com.example.cm.databinding.FragmentMeetupBinding;
 import com.example.cm.databinding.FragmentMeetupListBinding;
 import com.example.cm.ui.adapters.MeetupListAdapter;
-import com.example.cm.ui.dashboard.DashboardViewModel;
-import com.example.cm.ui.profile.ProfileViewModel;
 import com.example.cm.utils.Navigator;
 
-import timber.log.Timber;
-
-public class MeetupListFragment extends Fragment implements MeetupListAdapter.OnMeetupClickedListener {
+public class MeetupListFragment extends Fragment {
 
     private FragmentMeetupListBinding binding;
-    private MeetupListViewModel meetupListViewModel;
     private Navigator navigator;
     private MeetupListAdapter meetupListAdapter;
 
+    @SuppressLint("NotifyDataSetChanged")
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
 
         binding = FragmentMeetupListBinding.inflate(inflater, container, false);
         navigator = new Navigator(requireActivity());
         setHasOptionsMenu(true);
+
         initUi();
         initViewModel();
 
         return binding.getRoot();
     }
 
+    @SuppressLint("NotifyDataSetChanged")
     private void initViewModel() {
-        meetupListViewModel = new ViewModelProvider(this).get(MeetupListViewModel.class);
-        meetupListViewModel.getMeetups().observe(getViewLifecycleOwner(), meetups -> {
-            if(meetups == null){
-                return;
-            }
-            meetupListAdapter.setMeetups(meetups);
+        MeetupListViewModel meetupListViewModel = new ViewModelProvider(this).get(MeetupListViewModel.class);
+        meetupListViewModel.getLiveMeetupData().observe(getViewLifecycleOwner(), meetups -> {
```

Comment:
Maybe rename getLiveMeetupData to getMeetups (Datatype not needed in method name)